src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -158,6 +158,47 @@ function mpas_kd_construct(points, ndims) result(tree)
 
    end function mpas_kd_construct
 
+   !***********************************************************************
+   !
+   !  routine break_tie
+   !
+   !> \brief   Break a tie for two n-dim points
+   !> \author  Miles A. Curry
+   !> \date    07/07/20
+   !> \details
+   !> Compare 1..n dimensions of p1 and p2 and return -1 if p1(n) is less than
+   !> p2(n) and return 1 if p1(n) is greater than p2(n). If p1(n) and p2(n) are
+   !> equal than the some comparison will be done on p1(n+1) and p2(n+1). If 
+   !> p1(:) and p2(:) are equal across all n (i.e. they are the exactly the same 
+   !> point), then 0 will be returned.
+   !
+   !-----------------------------------------------------------------------
+   function break_tie(p1, p2) result(tie)
+
+      implicit none
+
+      ! Input Variables
+      type (mpas_kd_type), intent(in) :: p1
+      type (mpas_kd_type), intent(in) :: p2
+      integer :: tie
+
+      integer :: i
+
+      tie = 0
+      do i = 1, size(p1 % point(:))
+          ! Go to the next dimension for comparision if the two points are equal on a dimension
+          if (p1 % point(i) == p2 % point(i)) then
+             cycle
```

Comment:
In my opinion, it's a little awkward to have a `cycle` statement here. Wouldn't the following loop without a `cycle` statement work?
```fortran
      do i = 1, size(p1 % point(:))                                                                                                  
          if (p1 % point(i) < p2 % point(i)) then                                                                                    
              tie = -1                                                                                                               
              return                                                                                                                 
          else if (p1 % point(i) > p2 % point(i)) then                                                                               
              tie = 1                                                                                                                
              return                                                                                                                 
          end if                                                                                                                     
      end do 
```

Reply:
Also, I'm not sure whether the loop as written (with the `cycle` statement) works as expected. For example, if I define two points:
```fortran
      p1 % point(:) = [1.0, 1.0, 0.0]                                            
      p2 % point(:) = [0.0, 0.0, 1.0]    
```
the `break_tie` function returns -1 (i.e., p1 < p2), which isn't what I was expecting. However, it could be that I've misunderstood the expected behavior. Ultimately, as long as the function works deterministically, we should be fine; but, it's always nice when the function behavior matches what most people would intuitively expect.

Reply:
@mgduda I think the portion of code is also a bit awkward, particularly I don't like the double if statements. I like your implementation better and I'll switch it over to that.

Too your second comment, I think -1 was being returned as there were a few return statements missing. When I wrote my unit test, I wrote a test and figured it was enough, but in the back of my mind I knew I needed a few more, and it appears that thought was right, so I'll add a few more tests to make sure we've got good coverage.